src/main/java/duke/tools/TaskList.java

```diff
@@ -0,0 +1,202 @@
+        if (parts.length < 2) {
+            throw new DukeException("Which task do you want to mark as done?");
+        }
+        String index = parts[1];
+        int taskIndex = 0;
+        try {
+            taskIndex = Integer.parseInt(index) - 1;
+        } catch (NumberFormatException e) {
+            System.out.println("Please enter a valid index.");
+        }
+        if (taskIndex > taskList.size() || taskIndex < 0) {
+            throw new IndexOutOfBoundsException("Please enter a valid index.");
+        }
+        Task taskChanged = taskList.get(taskIndex);
+        String action = parts[0];
+        try {
+            if (action.equals("mark")) {
+                taskChanged.markDone();
+                System.out.println("Nice! I've marked this task as done:");
+            } else {
+                taskChanged.markUndone();
+                System.out.println("Nice! I've marked this task as undone:");
+            }
+            System.out.println(taskChanged);
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Method deletes task from taskList.
+     * @param task The instructions containing index of task to be deleted.
+     */
+    public void delete(String task) throws DukeException {
+        String[] segments = task.split(" ");
+        if (segments.length < 2) {
+            throw new DukeException("Which task do you want to delete?");
+        }
+        String index = segments[1];
+        int taskIndex;
+        try {
+            taskIndex = Integer.parseInt(index) - 1;
+        } catch (NumberFormatException e) {
+            throw new DukeException("Please enter a valid index."); //e.g. delete hi
+        }
+        if (taskIndex > taskList.size() || taskIndex < 0) {
+            throw new DukeException("Please enter a valid index.");
+        }
+        Task deletedTask = taskList.get(Integer.parseInt(index) - 1);
+        taskList.remove(deletedTask);
+
+        System.out.println(line);
+        System.out.println("Deleted the following task: ");
+        System.out.println(deletedTask);
+        System.out.println(line);
+    }
+
+    /**
+     * Method to print taskList.
+     */
+    public void printList() throws DukeException {
+        if (taskList.isEmpty()) {
+            throw new DukeException("You have no tasks in your list! Yay!");
+        } else {
+            int index = 1;
+            for (Task task : taskList) {
+                System.out.println(index + ". " + task);
+                index++;
+            }
+        }
+    }
+
+    /**
+     * Method to handle inputs.
+     * Entry point linking terminal to system.
+     */
+    public void handleInput() throws DukeException {
+        Scanner sc = new Scanner(System.in);
+        String task = sc.nextLine();
+        KeywordEnum keywordEnum = KeywordEnum.assign(task);
+
+        while (true) {
+            switch(keywordEnum) {
+                case LIST:
```

Comment:
Iirc the coding style used in this course suggests for no indentation for `case` clauses 🤔